src/score_hv/harvesters/daily_bfg.py

```diff
@@ -141,107 +231,154 @@ def get_data(self):
-            
-            for j, statistic in enumerate(self.config.get_stats()):
-                """ The second nested loop iterates through each requested 
-                    statistic
-                """
-                if statistic == 'mean':
-                    value = expected_value
-                elif statistic == 'variance':
-                    value = stats_utils.area_weighted_variance(
-                                                temporal_means,
-                                                gridcell_area_weights,        
-                                                expected_value=expected_value)
-                elif statistic == 'maximum':
-                    value = np.ma.max(temporal_means)
-                
-                elif statistic == 'minimum':
-                    value = np.ma.min(temporal_means)
-                
-                harvested_data.append(HarvestedData(
-                                    self.config.harvest_filenames,
-                                    statistic, 
-                                    variable,
-                                    np.float32(value),
-                                    units,
-                                    dt.fromisoformat(median_cftime.isoformat()),
-                                    longname))
-        
+                   msg=f"No longitude values found within the specified range of {east_lon} and {west_lon}."
+                   raise KeyError(msg) 
+
+                var_data=variable_data.isel(time=slice(None), \
+                                            grid_yt=slice(lat_start_index,lat_end_index+1), \
+                                            grid_xt=slice(east_start_index,west_end_index))
+                weights=gridcell_area_data['area'].isel(grid_yt=slice(lat_start_index,lat_end_index + 1), \
+                                                        grid_xt=slice(east_start_index, west_end_index))
+                latitudes=xr_dataset['grid_yt']
+                longitudes=xr_dataset['grid_xt']
+                num_lat = len(weights['grid_yt'])
+                num_lon = len(weights['grid_xt'])
+                value  = np.ma.masked_invalid(var_data.mean(dim='time',skipna=True))
+                temporal_means.append(value)                  
+                var_stats_instance.calculate_requested_statistics(var_data,weights,value)
+               
+            for iregion in range(num_regions): 
+                for j, statistic in enumerate(self.config.get_stats()):
+                       """ The second nested loop iterates through each requested 
+                           statistic and regions if the user has requested geographical
+                           regions..
+                       """
+                       if statistic == 'mean':
+                          themeans=var_stats_instance.weighted_averages[iregion] 
+                          value=themeans 
```

Comment:
why not just set value directly: value=var_stats_instance.weighted_averages[iregion] ?

Reply:
same comment for the variables below. I think it's cleaner to set it directly